src/pykeen/hpo/hpo.py

```diff
@@ -812,10 +811,14 @@ def _set_study_dataset(
         or isinstance(dataset, Dataset)
         or (isinstance(dataset, type) and issubclass(dataset, Dataset))
     ):
+        if not(training is None and testing is None and validation is None):
+            raise ValueError("Cannot specify dataset and training, testing and validation")
         dataset_name = get_dataset(dataset=dataset).get_normalized_name()
         study.set_user_attr('dataset', dataset_name)
     else:
-        study.set_user_attr('dataset', USER_DEFINED_CODE)
-        study.set_user_attr('training', training if isinstance(training, str) else USER_DEFINED_CODE)
-        study.set_user_attr('testing', testing if isinstance(testing, str) else USER_DEFINED_CODE)
-        study.set_user_attr('validation', validation if isinstance(validation, str) else USER_DEFINED_CODE)
+        if isinstance(training, str):
```

Comment:
isn't it helpful to know that these aren't included in the HPO becuase they are user defined? what's the rationale behind changing this

Reply:
The issue is that for the final model training, the dataset instation will fail, because both `dataset `and `training-validation-and testing` are defined. The pipeline configuration should only contain one of both.

Reply:
do you have an example that illustrates the issue?